builtin/game/item.lua

```diff
@@ -205,6 +207,25 @@ function core.get_node_drops(nodename, toolname)
 	return got_items
 end
 
+local non_player_name = ""
+local function user_name(user)
+	return user and user:is_player() and user:get_player_name() or ""
+end
+
+local function is_protected(pos, name)
+	return core.is_protected(pos, name) and
+		not minetest.check_player_privs(name, "protection_bypass")
+end
+
+-- Returns a logging function. For nil names, does not log.
+local function make_log(name)
+	if name ~= non_player_name then
```

Comment:
Comment:  \* For empty names

 Do we need `non_empty_player`? As we already have this empty string documented as non-player player name it would be fine to simply do `if name ~= "" then`.